src/Http/Http/src/Internal/ResponseCookies.cs

```diff
@@ -88,6 +88,57 @@ public void Append(string key, string value, CookieOptions options)
             Headers[HeaderNames.SetCookie] = StringValues.Concat(Headers[HeaderNames.SetCookie], cookieValue);
         }
 
+        /// <inheritdoc />
+        public void Append(IDictionary<string, string> keyValuePairs, CookieOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            // SameSite=None cookies must be marked as Secure.
+            if (!options.Secure && options.SameSite == SameSiteMode.None)
+            {
+                if (_logger == null)
+                {
+                    var services = _features.Get<Features.IServiceProvidersFeature>()?.RequestServices;
+                    _logger = services?.GetService<ILogger<ResponseCookies>>();
+                }
+
+                if (_logger != null)
+                {
+                    foreach (var keyValuePair in keyValuePairs)
+                    {
+                        Log.SameSiteCookieNotSecure(_logger, keyValuePair.Key);
+                    }
+                }
+            }
+
+            var setCookieHeaderValue = new SetCookieHeaderValue(string.Empty)
+            {
+                Domain = options.Domain,
+                Path = options.Path,
+                Expires = options.Expires,
+                MaxAge = options.MaxAge,
+                Secure = options.Secure,
+                SameSite = (Net.Http.Headers.SameSiteMode)options.SameSite,
+                HttpOnly = options.HttpOnly
+            };
+
+            var cookierHeaderValue = setCookieHeaderValue.ToString()[1..];
+            var cookies = new string[keyValuePairs.Count];
+            var position = 0;
+
+            foreach (var keyValuePair in keyValuePairs)
+            {
+                cookies[position] = $"{keyValuePair.Key}={keyValuePair.Value}{cookierHeaderValue}";
```

Comment:
Could you use a string concat (or a `StringBuilder` here? No reason to interpolate this.

Reply:
String concat is better than StringBuilder for simple scenarios like this.

Reply:
You've skipped the escaping step from the other API:
https://github.com/dotnet/aspnetcore/blob/00b551ec3ca7b50cbfe7fc0cfc91fe9e51b37b99/src/Http/Http/src/Internal/ResponseCookies.cs#L73-L75

Reply:
- I have changed to use `string.concat`
- yep, I forgot the escaping part, I have already added it back

Thanks!

Reply:
FWIW the C# compiler [creates string.Concat](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ+WRjgAykCyAFC+wNYwAngBpybUgDcAhgBsArjFG9SACxhSAJjCgA1WQoCU9RiYC8APlIASAEQ0BggL6ma0+TEc01m7XveObAG5CRyA=) for this on our behalfes. So you can use the most readable code.